app/models/library_location.rb

```diff
@@ -2,12 +2,13 @@
 #  Class to handle configuration and logic around library codes and labels
 ###
 class LibraryLocation
-  attr_reader :library, :location
+  attr_reader :request, :library, :location
 
   include Scannable
   include Mediateable
 
   def initialize(request)
+    @request = request
```

Comment:
I wouldn't hold up merging the PR over this, but I'm a little bothered by coupling a request and a location like this. Maybe a future refactor could make `Request` support `scannable_item?`, and a `LibraryLocation` support `scannable_location?` (or whatever the word for origin library + origin location actually is..).